src/session-pool.js

```diff
@@ -550,399 +508,169 @@ SessionPool.prototype.createTransaction_ = function(session, options) {
 };
 
 /**
- * Attempts to delete a session, optionally creating a new one of the same type
- * if the pool is still open and we're under the configured min value.
- *
- * @private
- *
- * @fires SessionPool#destroy
- * @fires SessoinPool#error
- * @param {Session} session The session to delete.
- * @returns {Promise}
- */
-SessionPool.prototype.destroySession_ = function(session) {
-  var self = this;
-
-  this.spliceSession_(session);
-
-  if (!this.available()) {
-    this.emit('empty');
-  }
-
-  if (this.isOpen && this.needsFill_()) {
-    this.fill();
-  }
-
-  return this.requestQueue_
-    .add(function() {
-      return session.delete();
-    })
-    .catch(function(err) {
-      self.emit('error', err);
-    });
-};
-
-/**
- * Deletes idle sessions that exceed the maxIdle configuration.
+ * Attemps to create a write session
  *
- * @private
  *
- * @returns {Promise}
+ * @return {Promise<Session>}
  */
-SessionPool.prototype.evictIdleSessions_ = function() {
-  var self = this;
-  var evicted = [];
-
-  var maxIdle = this.options.maxIdle;
-  var min = this.options.min;
-  var size = this.size();
-
-  var idle = this.getIdleSessions_();
-  var count = idle.length;
-
-  while (count-- > maxIdle && size - evicted.length > min) {
-    evicted.push(idle.pop());
-  }
-
-  return Promise.all(
-    evicted.map(function(session) {
-      return self.destroySession_(session);
-    })
-  );
+SessionPool.prototype.createWriteSession = function() {
+  const self = this;
+  return self.requestQueue_.add(() => {
+    const session = self.session_();
+
+    return session
+      .create()
+      .then(function() {
+        session.type = READWRITE;
+        return self.createTransaction_(session);
+      })
+      .then(() => session)
+      .catch(err => {
+        self.destroySession(session);
+        throw err;
+      });
+  });
 };
 
 /**
- * Retrieves a list of all the idle sessions.
+ * Attemps to create a read session
  *
- * @private
  *
- * @returns {Session[]}
+ * @return {Promise<Session>}
  */
-SessionPool.prototype.getIdleSessions_ = function() {
-  var idlesAfter = this.options.idlesAfter * 60000;
-  var sessions = this.reads_.concat(this.writes_);
-
-  return sessions.filter(function(session) {
-    return Date.now() - session.lastUsed >= idlesAfter;
+SessionPool.prototype.createReadSession = function() {
+  const self = this;
+  return self.requestQueue_.add(() => {
```

Comment:
We're missing a test to ensure the request is queued.

Reply:
Added the test case